### PR TITLE
Feature/find job page

### PR DESCRIPTION
**Pull request by SergKukh**
What was done before first review:
created jobs api and added fetching jobs from backend for Jobs Find page
added pagination for Find Jobs page
delete max items validation for selects in filter component
refactor types in redux

What's new:
fixed loader issue
finished filter logic in Find Jobs page
added search to Find Jobs page

files edited after 1st review:
apps/freelance/src/pages/FindJobs/index.tsx
apps/freelance/src/pages/FindJobs/utils.ts
apps/freelance/src/redux/services/jobsApi.ts
package.json

image

image

image

image

image

### CODE REVIEW

File: apps/freelance/src/pages/FindJobs/index.tsx [hunk 1]
```diff
@@ -36,25 +43,36 @@ const FindJobs = () => {
...
-              rate={item.rate}
-            />
-          </List.Item>
-        )}
-      />
-
-      <StyledPagination />
+      {isLoading ? (
+        <Spin size="large" />
+      ) : (
```
Comment (lyudad) [inline apps/freelance/src/pages/FindJobs/index.tsx]: loader better to move to the wrapper, you avoid ternary operator in each component

Comment (SergKukh) [reply apps/freelance/src/pages/FindJobs/index.tsx]: @lyudad refactored with loading prop in List component

File: apps/freelance/src/pages/FindJobs/index.tsx [hunk 2]
```diff
@@ -1,60 +1,101 @@
...
+    setFilterPayload(getFilterParams(filterData));
+  };
+
+  const onChangePagination: PaginationProps['onChange'] = page => {
+    setOffset((page - 1) * fetchLimit);
+  };
+
+  const onSearch = (value: string) => {
+    setSearch(value.trim());
   };
```
Comment (lyudad) [inline apps/freelance/src/pages/FindJobs/index.tsx]: too many logic, move to custom hook

File: apps/freelance/src/pages/FindJobs/index.tsx [hunk 3]
```diff
@@ -1,60 +1,101 @@
...
+} from './styles';
+import { getFilterParams } from './utils';
 
 const FindJobs = () => {
   const [filtersVisibility, setFiltersVisibility] = useState<boolean>(false);
+  const [offset, setOffset] = useState<number>(0);
+  const [filterPayload, setFilterPayload] = useState<GetJobParams>();
+  const [search, setSearch] = useState<string>();
   const { t } = useTranslation();
+  const { Search } = Input;
```
Comment (lyudad) [inline apps/freelance/src/pages/FindJobs/index.tsx]: this initialise with each rerender, we dont need to have it inside the component

File: apps/freelance/src/pages/FindJobs/index.tsx (hunk 2 above)
Comment (SergKukh) [reply apps/freelance/src/pages/FindJobs/index.tsx]: @lyudad moved to useFindJobs hook

File: apps/freelance/src/pages/FindJobs/index.tsx (hunk 3 above)
Comment (SergKukh) [reply apps/freelance/src/pages/FindJobs/index.tsx]: @lyudad fixed

**Review by lyudad — APPROVED** [review]

**Pull request merged**